### PR TITLE
Feature/send transaction layer 1 (#1843)

### LINKED ISSUE #1843: Wallet - Send transaction, layer 1
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

Implementing 2-5 from https://docs.google.com/document/d/13eItUUqlfkRpx7iSutPRvThzTSrhirnKBcgxhOD3m6g/edit

### PR DESCRIPTION
**Pull request by flexsurfer**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)
fixes #1843

1)Implemented send transaction
2)implemented amount validation for request
3)fixed title for toolbar2 for ios (centered title)
4)fixed toolbar2 colors - wallet
5)fixed statusbar colors - wallet
6)added letsubs in profile
7)moved choose-recipient from send view (@angusiguess)

send-transaction-android
send-transaction-enter-password-android
send-transaction-enter-password-ios
send-transaction-enter-wrong-password-android
send-transaction-enter-wrong-password-ios
send-transaction-ios
send-transaction-not-valid-android
send-transaction-not-valid-ios
send-transaction-sent-android
send-transaction-sent-ios

[comment]: # (PRs will only be accepted if squashed into single commit.)
status: ready

### CODE REVIEW

File: src/status_im/ui/screens/wallet/components/views.cljs [hunk 1]
```diff
@@ -3,27 +3,45 @@
...
-            :placeholder-text-color "#ffffff66"
-            :selection-color        :white
-            :style                  styles/text-input
-            :on-focus               #(do (reset! active? true)
-                                         (when on-focus (on-focus)))
-            :on-blur                #(do (reset! active? false)
-                                         (when on-blur (on-blur)))}
-           (dissoc input-options :on-focus :on-blur))]]])))
+            [reagent.core :as reagent]
+            [status-im.components.icons.vector-icons :as vi]
```
Comment (yenda) [inline src/status_im/ui/screens/wallet/components/views.cljs]: we shouldn't abbreviate namespaces aliases

File: src/status_im/ui/screens/wallet/components/views.cljs [hunk 2]
```diff
@@ -3,27 +3,45 @@
...
+  (views/letsubs [bottom-value (animation/create-value 16)
+                  opacity-value (animation/create-value 0)]
+    {:component-did-mount (animations/animate-tooltip bottom-value opacity-value)}
+    [react/view styles/tooltip-container
+     [react/animated-view {:style (merge (styles/tooltip-animated bottom-value opacity-value) style)}
+      [react/view styles/tooltip-text-container
+       [react/text {:style styles/tooltip-text} label]]
+      [vi/icon :icons/tooltip-triangle {:color :white :style styles/tooltip-triangle}]]]))
+
+(defn amount-input []
```
Comment (yenda) [inline src/status_im/ui/screens/wallet/components/views.cljs]: why all the inline styling in that view ? (not that I am against it sometimes it feels like overkill that we have the non-reusable styling in separate file)

File: src/status_im/ui/screens/wallet/request/events.cljs [hunk 3]
```diff
@@ -10,4 +10,11 @@
...
                    {:contact contact
-                    :amount (:amount request-transaction)}]]}))
\ No newline at end of file
+                    :amount (:amount request-transaction)}]]}))
+
+(handlers/register-handler-fx
+  :wallet-validate-request-amount
+  (fn [{{:keys [web3] :wallet/keys [request-transaction] :as db} :db} _]
+    (let [amount (:amount request-transaction)
+          error (wallet.db/getAmountValidationError amount web3)]
```
Comment (yenda) [inline src/status_im/ui/screens/wallet/request/events.cljs]: wouldn't it be clearer if getAmountValidationError (why is it camel and not kebab case btw ?) returned nil for the good cases ?

File: src/status_im/ui/screens/wallet/db.cljs [hunk 4]
```diff
@@ -15,3 +19,20 @@
...
+      (or (nil? amount) (= amount "") (= amount "0") (re-matches #"0[,.]0*$" amount))
+      ""
+
+      (or (js/isNaN (js/parseFloat amount'))
+          (try (when (<= (.toWei web3 amount' "ether") 0) true)
+               (catch :default err true)))
+      (i18n/label :t/validation-amount-invalid-number)
+
+      (and (= (count amount-splited) 2) (> (count (last amount-splited)) 18))
+      (i18n/label :t/validation-amount-is-too-small)
```
Comment (yenda) [inline src/status_im/ui/screens/wallet/db.cljs]: I am not sure I get this one ? When there is more than 18 decimals the amount is considered too small ?

**Review by yenda — COMMENTED** [review]
I still need to review the events part, I'll do that by working on my PR off that branch

File: src/status_im/ui/screens/wallet/db.cljs (hunk 4 above)
Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/db.cljs]: right

Comment (yenda) [reply src/status_im/ui/screens/wallet/db.cljs]: but why is it too small ? maybe too precise ? can't it be 1000.00000000000000...1 or even 0.90000000000000...1 ?

Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/db.cljs]: yes, you're right, if you're talking about keyword name, again i just take it from bot https://github.com/status-im/status-react/blob/develop/resources/js/bots/wallet/translations.js#L32

Comment (yenda) [reply src/status_im/ui/screens/wallet/db.cljs]: ah ok I see, can you rename it too-precise then ?

Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/db.cljs]: yes,sure,thanks

File: src/status_im/ui/screens/wallet/request/events.cljs (hunk 3 above)
Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/request/events.cljs]: done, thanks

File: src/status_im/ui/screens/wallet/components/views.cljs (hunk 2 above)
Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/components/views.cljs]: could you elaborate? what does it mean `why all the inline styling in that view `

File: src/status_im/ui/screens/wallet/components/views.cljs [hunk 5]
```diff
@@ -3,27 +3,45 @@
...
+  (let [active? (reagent/atom false)]
+    (fn [& [{:keys [input-options style error]}]]
+      (let [{:keys [on-focus on-blur]} input-options]
+        [react/view {:flex 1}
```
Comment (yenda) [inline src/status_im/ui/screens/wallet/components/views.cljs]: for instance this is equivalent to styles/flex

File: src/status_im/ui/screens/wallet/components/views.cljs [hunk 6]
```diff
@@ -33,6 +51,25 @@
...
 
+(defn choose-recipient [{:keys [address name on-press style]}]
+  (let [address? (and (not (nil? address)) (not= address ""))]
+    [react/touchable-highlight {:on-press on-press}
+     [react/view
+      [react/text {:style styles/label} (i18n/label :t/recipient)]
+      [react/view (merge styles/recipient-container
+                         style)
+       (when name
+         [react/view {:padding-right 6}
```
Comment (yenda) [inline src/status_im/ui/screens/wallet/components/views.cljs]: here the style is inline, and so on...

File: src/status_im/ui/screens/wallet/components/views.cljs (hunk 2 above)
Comment (yenda) [reply src/status_im/ui/screens/wallet/components/views.cljs]: I pointed a few examples, note that I don't see it as a problem actually sometimes I wonder why we don't do inline in the first place

File: src/status_im/ui/screens/wallet/components/views.cljs (hunk 5 above)
Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/components/views.cljs]: just missed this

File: src/status_im/ui/screens/wallet/components/views.cljs (hunk 6 above)
Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/components/views.cljs]: just missed this

File: src/status_im/ui/screens/wallet/send/transaction_sent/views.cljs [hunk 7]
```diff
@@ -0,0 +1,38 @@
+(ns status-im.ui.screens.wallet.send.transaction-sent.views
+  (:require-macros [status-im.utils.views :refer [defview letsubs]])
+  (:require [status-im.components.react :as react]
+            [status-im.components.status-bar :as status-bar]
+            [status-im.components.icons.vector-icons :as vi]
```
Comment (yenda) [inline src/status_im/ui/screens/wallet/send/transaction_sent/views.cljs:5]: I missed this one (there are other but on ns that need to be refactored)

File: src/status_im/ui/screens/wallet/send/views.cljs [hunk 8]
```diff
@@ -13,77 +12,113 @@
...
             [status-im.ui.screens.wallet.send.styles :as send.styles]
             [status-im.components.icons.vector-icons :as vector-icons]
-            [reagent.core :as r]))
+            [reagent.core :as r]
+            [status-im.utils.platform :as platform]
+            [status-im.ui.screens.wallet.request.styles :as request.styles]
+            [status-im.ui.screens.wallet.send.styles :as styles]
+            [status-im.ui.screens.wallet.components.views :as components]
+            [status-im.components.styles :as components.styles]
+            [status-im.components.icons.vector-icons :as vi]
```
Comment (yenda) [inline src/status_im/ui/screens/wallet/send/views.cljs:21]: here too

**Pull request merged**